src/object/obj_tx.c

```diff
@@ -1548,6 +1578,35 @@ dc_tx_commit_prepare(struct dc_tx *tx, tse_task_t *task)
 	if (shard_tgts == NULL)
 		D_GOTO(out, rc = -DER_NOMEM);
 
+	mbs = dcsh->dcsh_mbs;
+	mbs->dm_flags = DMF_CONTAIN_LEADER;
+
+	/* For the case of modification(s) within single RDG,
+	 * elect leader as standalone modification case does.
+	 */
+	if (mod_grp_cnt == 1) {
+		i = dc_tx_leftmost_req(tx, true);
+		dcsr = &tx->tx_req_cache[i];
+		obj = dcsr->dcsr_obj;
+
+		grp_idx = obj_dkey2grpidx(obj, dcsr->dcsr_dkey_hash,
+					  tx->tx_pm_ver);
+		if (grp_idx < 0)
+			D_GOTO(out, rc = grp_idx);
+
+		i = pl_select_leader(obj->cob_md.omd_id,
+				     grp_idx * obj->cob_grp_size,
+				     obj->cob_grp_size, false,
+				     obj_get_shard, obj);
+		if (i < 0)
+			D_GOTO(out, rc = i);
+
+		leader_oid.id_pub = obj->cob_md.omd_id;
+		leader_oid.id_shard = i;
+		leader_dtrg_idx = obj_get_shard(obj, i)->po_target;
+		mbs->dm_flags |= DMF_MODIFY_SRDG;
```

Comment:
ok, so both DMF_CONTAIN_LEADER and DMF_MODIFY_SRDG can set?
seems not same as what in the comment "For stand-alone modification, leader information is not stored inside MBS as optimization"?

Reply:
Yes, DMF_CONTAIN_LEADER and DMF_MODIFY_SRDG can be set at the same time if the distributed transaction only modifies the targets in one redundancy group. As explained above, "stand-along-modification" means non-TX update/punch.